core/src/mindustry/core/UI.java

```diff
@@ -32,8 +32,29 @@
 import static arc.scene.actions.Actions.*;
 import static mindustry.Vars.*;
 
+import java.util.Arrays;
+
 public class UI implements ApplicationListener, Loadable{
-    public static String billions, millions, thousands;
+    private static final long[] power10s = {
```

Comment:
With this many values and this much logic, I think it would be better idea to move *all* the formatting logic into a separate class, say, `NumFormat`, and have `UI` delegate `formatAmount` to a static method in that class for compatibility.